eis_qgis_plugin/wizard/eda/plots/parallel_coordinates_raster.py

```diff
@@ -0,0 +1,117 @@
+from typing import List
+
+import matplotlib.colors as mcolors
+import matplotlib.pyplot as plt
+import numpy as np
+from qgis.core import QgsMapLayerProxyModel, QgsRasterLayer
+from qgis.gui import QgsMapLayerComboBox
+from qgis.PyQt.QtWidgets import QGroupBox, QSizePolicy, QWidget
+
+import eis_qgis_plugin.libs.seaborn as sns
+from eis_qgis_plugin.qgis_plugin_tools.tools.resources import load_ui
+from eis_qgis_plugin.utils.layer_data_table import LayerDataTable
+from eis_qgis_plugin.utils.misc_utils import check_raster_grids
+from eis_qgis_plugin.wizard.eda.plots.parallel_coordinates import EISWizardParallelCoordinatesPlot
+
+FORM_CLASS: QWidget = load_ui("eda/wizard_plot_parallel_coordinates_raster.ui")
+
+
+class EISWizardParallelCoordinatesRasterPlot(EISWizardParallelCoordinatesPlot, FORM_CLASS):
+    """
+    Class for EIS parallel coordinate plots with raster layers.
+
+    Initialized from a UI file. Responsible for updating widgets and
+    producing the plot.
+    """
+
+    def __init__(self, parent=None) -> None:
+        
+        # DECLARE TYPES
+        self.data_box: QGroupBox
+        self.color_selection: QgsMapLayerComboBox
+
+        # Initialize
+        self.collapsed_height = 270
+        super().__init__(parent)
+        
+        self.dtype = QgsRasterLayer
+
+        self.data_box.setSizePolicy(QSizePolicy.Preferred, QSizePolicy.Maximum)
+
+        self.data_layer_table = LayerDataTable(self, dtype=QgsRasterLayer)
+        self.data_box.layout().addWidget(self.data_layer_table)
+
+        self.color_selection.setFilters(QgsMapLayerProxyModel.RasterLayer)
+        self.color_selection.setAllowEmptyLayer(True)
+        self.color_selection.setCurrentIndex(0)
+
+        self.data_layer_table.size_changed.connect(self._update_size)
+
+        self.data_layer_table.add_row()
+
+
+    def _update_size(self, size_change: int):
+        self.setMinimumHeight(self.minimumHeight() + size_change)
+        self.setMaximumHeight(self.maximumHeight() + size_change)
+
+
+    def reset(self):
+        """Reset parameters to defaults."""
+        super().reset()
+
+        self.color_selection.setCurrentIndex(0)
+
+
+    def get_layers(self) -> List[QgsRasterLayer]:
+        return [
+            self.data_layer_table.cellWidget(row, 0).currentLayer() for row in range(self.data_layer_table.rowCount())
+        ]
+
+
+    def prepare_data(self):
+        # Get input values
+        rasters = self.get_layers()
+        raster_names = [raster.name() for raster in rasters]
+
+        # Check matching raster CRSs, cell sizes, pixel alignments, and bounds
+        check_raster_grids(rasters)
+            
+        # Get data as Numpy array
+        height = rasters[0].height()
+        width = rasters[0].width()
+        raster_data = np.empty((height * width, 0), dtype=np.float32)
+        for raster in rasters:
```

Comment:
Currently, the plot is not produced if nodata is present, since the empty Numpy array initialization does not take into account nodata cells that will be removed in `raster_layer_to_array`. This is one solution I found out while testing:
```suggestion
        # Initialize with first raster
        raster_data = self.raster_layer_to_array(rasters[0]).reshape(-1, 1)
        for raster in rasters[1:]:
```